FamliCareClientProject/famlicare-client-project
# Gymnarctos fix registration profile

## Changes proposed in this pull request:

- Enhanced error logging for file upload and presigned URL generation processes
- Added test routes to isolate and debug S3 operations
- Reviewed and updated CORS configuration to ensure proper client-server communication

## Why these changes are necessary:

We've been encountering issues with file uploads during user registration, specifically a 403 Forbidden error when attempting to generate a presigned URL. These changes aim to provide more detailed error information, allow for isolated testing of S3 operations, and ensure our CORS configuration is correct.

## Specific updates:

1. In `user.router.js`:
   - Added more detailed error logging in the presigned URL generation route
   - Improved error handling in the file upload route
   - Included error stack traces in API responses for better debugging

2. New test routes in `user.router.js`:
   - `/test-presigned-url`: Generates a presigned URL without authentication
   - `/test-upload`: Attempts a direct file upload to S3 from the server

3. In `server.js`:
   - Reviewed and updated CORS configuration
   - Now using an environment variable for the client URL in CORS config

## How to test these changes:

1. Attempt to register a new user with a profile picture
2. Check server logs for any detailed error messages if the upload fails
3. Use the new test routes to isolate S3 operations:
   - GET `/api/user/test-presigned-url`
   - GET `/api/user/test-upload`

## Additional notes:

- Make sure all necessary environment variables are set, including AWS credentials and S3 bucket name
- These changes do not alter the core functionality but aim to provide better debugging capabilities

Please review and test thoroughly before merging.
